includes/admin/load.php

```diff
@@ -204,137 +207,142 @@ function perflab_dismiss_wp_pointer_wrapper() {
  * Callback function to handle admin scripts.
  *
  * @since 2.8.0
+ * @since n.e.x.t Renamed to perflab_enqueue_features_page_scripts().
  */
-function perflab_enqueue_modules_page_scripts() {
-	wp_enqueue_script( 'updates' );
-
-	wp_localize_script(
-		'updates',
-		'_wpUpdatesItemCounts',
-		array(
-			'settings' => array(
-				'totals' => wp_get_update_data(),
-			),
-		)
-	);
-
+function perflab_enqueue_features_page_scripts() {
 	wp_enqueue_script( 'thickbox' );
 	wp_enqueue_style( 'thickbox' );
 
 	wp_enqueue_script( 'plugin-install' );
-
-	wp_enqueue_script(
-		'perflab-plugin-management',
-		plugin_dir_url( __FILE__ ) . 'js/perflab-plugin-management.js',
-		array( 'jquery' ),
-		'1.0.0',
-		array(
-			'in_footer' => true,
-			'strategy'  => 'defer',
-		)
-	);
 }
 
 /**
- * Callback function hooked to admin_action_perflab_activate_plugin to handle plugin activation.
+ * Callback for handling installation/activation of plugin.
  *
- * @since 2.8.0
+ * @since n.e.x.t
  */
-function perflab_activate_plugin() {
-	// Do not proceed if plugin query arg is not present.
-	if ( empty( $_GET['plugin'] ) ) {
-		return;
+function perflab_install_activate_plugin_callback() {
+	check_admin_referer( 'perflab_install_activate_plugin' );
+
+	require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-ajax-upgrader-skin.php';
+
+	if ( ! isset( $_GET['slug'] ) ) {
+		wp_die( esc_html__( 'Missing required parameter.', 'performance-lab' ) );
 	}
 
-	$plugin = sanitize_text_field( wp_unslash( $_GET['plugin'] ) );
+	$plugin_slug = sanitize_text_field( wp_unslash( $_GET['slug'] ) );
 
-	check_admin_referer( "perflab_activate_plugin_{$plugin}" );
+	if ( ! $plugin_slug ) {
+		wp_die( esc_html__( 'Invalid plugin.', 'performance-lab' ) );
+	}
+
+	$is_plugin_installed = ! isset( $_GET['file'] ) || ! $_GET['file'];
+
+	// Install the plugin if it is not installed yet.
+	if ( $is_plugin_installed ) {
```

Comment:
I think this logic is the wrong way around, given the variable name.
```suggestion
	$is_plugin_installed = isset( $_GET['file'] ) && $_GET['file'];

	// Install the plugin if it is not installed yet.
	if ( ! $is_plugin_installed ) {
```

Reply:
Perhaps simpler as:

```php
$is_plugin_installed = ! empty( $_GET['file'] );
```

Reply:
Still, seems like there should be some validation of this input here. And store the sanitized value in `$file`, as was done to store `$_GET['slug']` in `$plugin_slug`.

Reply:
Regarding the `empty`, I'd prefer not to use it, based on e.g. https://dev.to/klnjmm/never-use-empty-function-in-php-4pb0 or https://localheinz.com/articles/2023/05/10/avoiding-empty-in-php/. It's not a big deal either way, but the way it currently is written is more explicit.

+1 to sanitizing the input and storing it in `$file`!

Reply:
Sanitization added in https://github.com/WordPress/performance/pull/1091/commits/46357b03a53a67c1c404c8af99f62ec87a87f61b